test/export/note_sorter_spec.py

```diff
@@ -0,0 +1,122 @@
+from mamba import describe, it, context, before
+from unittest.mock import MagicMock
+from random import shuffle
+
+from test_utils.anki import mock_anki_modules
+
+mock_anki_modules()
+
+from crowd_anki.export.note_sorter import NoteSorter
+from crowd_anki.config.config_settings import ConfigSettings
+
+
+test_guids = ["aaabbbccc", "bbbcccddd", "cccdddeee", "dddeeefff", "eeefffggg", "fffggghhh"]
+test_flags = [0, 1, 2, 3, 4, 5]
+test_tags = ["adjectives", "directions", "interesting", "nouns", "verbs", "zzzzFinal"]
+test_notemodels = ["Default", "LL Noun", "LL Sentence", "LL Verb", "LL Word", "Zulu"]
+test_notemodelids = test_guids
+test_fields = test_tags
+
+
+class NoteSorterTester():
+    def __init__(self):
+        self.notes = self.setup_notes()
+
+        self.config = ConfigSettings(False)
+        self.config.export_deck_sort_reversed = False
+
+    def get_single_note_mock(self, i):
+        note = MagicMock()
+
+        note.anki_object.guid = test_guids[i]
+        note.anki_object.flags = test_flags[i]
+        note.anki_object.tags = test_tags[i]
+
+        note.anki_object._model = {}
+        note.anki_object._model.setdefault("name", test_notemodels[i])
+        note.anki_object._model.setdefault("crowdanki_uuid", test_notemodelids[i])
+
+        note.anki_object.fields = [test_fields[i], test_fields[i]]
+
+        return note
+
+    def setup_notes(self):
+        random_range = list(range(0, 6))
```

Comment:
tie this to size of data arrays/constant

Reply:
also with random things it's a good idea to print the current configuration you're working with (the shuffled number list in this case) to make any failures easier to reproduce

Reply:
Sure, but then it's just tied to one of them. I chose `test_guids`

Reply:
> also with random things it's a good idea to log the current configuration you're working with (the shuffled number list in this case) to make any failures easier to reproduce

Good call.

Reply:
As I asked in another comment, which level of logging would you do here?

Reply:
info is fine